lib/generators/boring/devise/jwt/install/install_generator.rb

```diff
@@ -0,0 +1,206 @@
+
+          abort
+        end
+
+        def verify_presence_of_devise_model
+          return if File.exist?("app/models/#{options[:model_name].underscore}.rb")
+
+          say "We couldn't find the #{options[:model_name]} model. Please provide correct model name", :red
+
+          abort
+        end
+
+        def add_devise_jwt_gem
+          say "Adding devise-jwt gem", :green
+          Bundler.with_unbundled_env do
+            run "bundle add devise-jwt"
+          end
+        end
+
+        def add_devise_jwt_config_to_devise_initializer
+          say "Adding devise-jwt config to devise initializer", :green
+          say "Value for jwt.secret will be used from `#{devise_jwt_secret_key}`. You can change this values if they don't match with your app.\n", :yellow
+          jwt_config = <<~RUBY
+            config.jwt do |jwt|
+              jwt.secret = #{devise_jwt_secret_key}
+              jwt.dispatch_requests = [
+                ['POST', %r{^/sign_in$}]
+              ]
+              jwt.revocation_requests = [
+                ['DELETE', %r{^/sign_out$}]
+              ]
+              jwt.expiration_time = #{options[:expiration_time_in_days]}.day.to_i
+            end
+          RUBY
+
+          inject_into_file "config/initializers/devise.rb",
+                            optimize_indentation(jwt_config, 2),
+                            before: /^end/
+        end
+
+        def configure_revocation_strategies
+          say "Configuring revocation strategies for #{options[:revocation_strategy]}", :green
+
+          case options[:revocation_strategy]
+          when "JTIMatcher"
+            configure_jti_matcher_strategy
+          when "Denylist"
+            configure_denylist_strategy
+          when "Allowlist"
+            configure_allowlist_strategy
+          end
+        end
+
+        private
+
+        def devise_jwt_secret_key
+          if options[:use_rails_credentials]
+            "Rails.application.credentials.devise_jwt_secret_key"
+          else
+            "ENV['DEVISE_JWT_SECRET_KEY']"
+          end
+        end
+
+        def configure_jti_matcher_strategy
+          model_name = options[:model_name].underscore
+          Bundler.with_unbundled_env do
+            run "bundle exec rails generate migration add_jti_to_#{model_name.pluralize}"
+          end
+          migration_content = <<~RUBY
+            add_column :users, :jti, :string, null: false
+            add_index :users, :jti, unique: true
+          RUBY
+
+          inject_into_file Dir["db/migrate/*_add_jti_to_#{model_name.pluralize}.rb"][0],
+                           optimize_indentation(migration_content, 4),
+                           after: /def change\n/,
+                           verbose: false
+
+          devise_model_content = <<~RUBY
+            include Devise::JWT::RevocationStrategies::JTIMatcher
+            devise :jwt_authenticatable, jwt_revocation_strategy: self
```

Comment:
Shouldn't we be appending these modules to existing list of modules instead of declaring new "devise" method invocation?

I have limited understanding of this as well. Won't this override previous set of declared modules e.g.

```
devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :validatable
```

Reply:
I believe it will not override, but I will tests to confirm whether it does. If it doesn't override, would it be acceptable to leave it as is?

Reply:
upon inspecting source code and doing manual test, it seems it will not override module but on every call to devise method they are including some default module which lead to duplicate include of default module and also they are maintaining the order while including module so might cause issue.

Reply:
If there is no existing devise method call, would this message suffice?
Successfully added the devise-jwt module to #{model_name} model. However, it looks like the devise module is missing from the #{model_name} model. Please configure the devise module to ensure everything functions correctly.

Reply:
Yes that message looks good